operator-common/src/main/java/io/strimzi/operator/common/Util.java

```diff
@@ -223,4 +222,35 @@ private static File store(String prefix, String suffix, KeyStore trustStore, cha
             throw e;
         }
     }
+
+    /**
+     * Logs environment variables into the regular log file.
+     */
+    public static void printEnvInfo() {
+        Map<String, String> env = new HashMap<>(System.getenv());
+        StringBuilder sb = new StringBuilder();
+
+        for (Map.Entry<String, String> entry: env.entrySet()) {
+            sb.append("\t").append(entry.getKey()).append(": ").append(maskPassword(entry.getKey(), entry.getValue())).append("\n");
+        }
+
+        LOGGER.info("Using config:\n" + sb.toString());
+    }
+
+    /**
+     * Gets environment variable, checks if it contains a password and in case it does it masks the output. It expects
+     * environment variables with passwords to contain `PASSWORD` in their name.
```

Comment:
Should we only be hiding passwords, might we want to hide apikeys or certificates as well?

Do let me know if I am wrong on this though...

Reply:
If so I propose we expand the list of masks

Reply:
We do not pass any private (or even public) keys this way. So we actually don't know how would the environment variables look like. I can add some generic strings - e.g. "PRIVATE_KEY`, `API_KEY` - but not sure they would match in the future.

Reply:
Makes sense to me, I was unsure if there were any and was just bringing it up in case there were any I was not aware of.

If you think it is unlikely that Strimzi will ever pass such values via Envars then I think you're right it is not needed, I do wonder though, if users are providing their own values for their own envars, e.g. in custom Connect images where they can configure extra envars for their own use cases might it make sense that we cover the hiding of such values by adding some extra generic strings like the ones you suggested above and briefly doc that these will be masked in the Envar list object that is used in the CRDs?

(Happy to not have it added though and only add it if a user requests such an enhancement)

Reply:
> If you think it is unlikely that Strimzi will ever pass such values via Envars then I think you're right it is not needed

I'm not sure I would call it unlikely. But until we have them, we do not have any name to filter against. So we just guess how our future self would name them. Or we should add them here hen we start using them.

> if users are providing their own values for their own envars, e.g. in custom Connect images where they can configure extra envars for their own use cases might it make sense that we cover the hiding of such values by adding some extra generic strings like the ones you suggested above and briefly doc that these will be masked in the Envar list object that is used in the CRDs?

This covers only the operators. Not things like Kafka Connect. Users can provide custom variables here as well, but there are no obvious plugable interfaces. So make any sense of passing something secret, they would need to modify the project quite significantly. Also, we again do not know the names, so it is hard to mask them.

So I would probably leave it out as it is.

Reply:
Thanks for clarifying, happy to approve :)

Reply:
Thanks 👍